samples/cpp/warpPerspective_demo.cpp

```diff
@@ -0,0 +1,231 @@
+            cout << "size rows and cols " << H.rows << " " << H.cols << endl;
+
+            Mat A(3, 4, CV_64F); //3xN, P1
+            Mat B(3, 4, CV_64F); //3xN, P2
+            //B = H*A  (P2 = h(P1))
+
+            for (int i = 0; i < 4; ++i)
+            {
+                A.at< double>(0, i) = P1[i].x;
+                A.at< double>(1, i) = P1[i].y;
+                A.at< double>(2, i) = 1;
+
+                B.at< double>(0, i) = P2[i].x;
+                B.at< double>(1, i) = P2[i].y;
+                B.at< double>(2, i) = 1;
+            }
+
+            cout << "a" << endl << A << endl;
+            cout << "b" << endl << B << endl;
+            Mat HA = H*A;
+
+            for (int i = 0; i < 4; ++i)
+            {
+                HA.at< double>(0, i) /= HA.at< double>(2, i);
+                HA.at< double>(1, i) /= HA.at< double>(2, i);
+                HA.at< double>(2, i) /= HA.at< double>(2, i);
+            }
+
+            cout << "HA" << endl << HA << endl;
+        }
+    }
+
+}
+
+Mat warping( Mat image, Size warped_image_size, vector< Point2f> P1, vector< Point2f> P2 )
+{
+    PointOrderbyCorner(P1);
+    //get homography
+    Mat H = findHomography(P1, P2);
+
+    //warping
+    Mat warped_image;
+    warpPerspective(image, warped_image, H, warped_image_size);
+    return warped_image;
+}
+
+void PointOrderbyCorner(vector< Point2f> inPoints)
+{
+    vector< pair< float, float> > s_point;
+    for (int i = 0; i< 4; ++i)
+        s_point.push_back(make_pair(inPoints[i].x, inPoints[i].y));
+
+    //sort
+    sort(s_point.begin(), s_point.end(), sorter_func());
+
+    if (s_point[0].first < s_point[1].first)
+    {
+        inPoints[0].x = s_point[0].first;
+        inPoints[0].y = s_point[0].second;
+
+        inPoints[1].x = s_point[1].first;
+        inPoints[1].y = s_point[1].second;
+    }
+    else
+    {
+        inPoints[0].x = s_point[1].first;
+        inPoints[0].y = s_point[1].second;
+
+        inPoints[1].x = s_point[0].first;
+        inPoints[1].y = s_point[0].second;
+    }
+
+    if (s_point[2].first > s_point[3].first)
+    {
+        inPoints[2].x = s_point[2].first;
+        inPoints[2].y = s_point[2].second;
+
+        inPoints[3].x = s_point[3].first;
+        inPoints[3].y = s_point[3].second;
+    }
+    else
+    {
+        inPoints[2].x = s_point[3].first;
+        inPoints[2].y = s_point[3].second;
+
+        inPoints[3].x = s_point[2].first;
+        inPoints[3].y = s_point[2].second;
+    }
+}
+
+static void onMouse(int event, int x, int y, int, void*)
+{
+    // Action when left button is clicked
+    for (int i = 0; i < 4; ++i)
+    {
+        if ((event == EVENT_LBUTTONDOWN) & ((abs(roi4point[i].x - x) < 10)) & (abs(roi4point[i].y - y) < 10))
+        {
+            selected_corner_index = i;
+            dragging = true;
+        }
+    }
+
+    if (event == EVENT_LBUTTONUP)
+    {
+        dragging = false;
+    }
+
+    if (event == EVENT_RBUTTONUP)
+    {
+        roi4point[0].x = 365;
```

Comment:
These hard-coded values look strange and for some images are not suitable. It's better to use relative quantities, depending on image width/height.